### DISCUSSION TITLE
Does everything go through Kong?

### DISCUSSION THREAD

**Discussion by idc77** [Kong/kong#10283] [Help]
I stumbled upon Kong while browsing supabase docs.
It says it uses Kong as an API-Gateway.

API-Gateway means for me that everything goes through Kong.
Kong is mostly LUA as far as I can tell.

When you have slow LUA on the front, and fast Go or Rust or C++ or Java on the back,
how does that work in terms of performance?
Wouldn't Kong bottleneck everything?

**Comment by chronolaw** [comment]
This blog may be helpful. https://konghq.com/blog/why-kong-is-the-best-api-gateway

**Reply by idc77** [reply]
That's a marketing babble post. no.

**Reply by hanshuebner** [reply]
The proxying itself is done by nginx.  Kong does not sit in the data path, but is only called - by the way of OpenResty - at the required points in the processing.

**Comment by ghost** [comment]
> When you have slow LUA on the front, and fast Go or Rust or C++ or Java on the back,

This issue is addressed in two points:

- Kong is built on OpenResty, which leverages LuaJIT to execute Lua code, providing a performance close to C.
- If code written in Lua is blocking, it will unavoidably cause delays, so when writing Lua code for OpenResty it is necessary to try to avoid such blocking code.
